genie-web/src/main/java/com/netflix/genie/web/agent/launchers/AgentLauncher.java

```diff
@@ -19,14 +19,15 @@
 
 import com.netflix.genie.web.dtos.ResolvedJob;
 import com.netflix.genie.web.exceptions.checked.AgentLaunchException;
+import org.springframework.boot.actuate.health.HealthIndicator;
 
 /**
  * A interface which implementations will launch instances of an agent in some manner in order to run a job.
  *
  * @author tgianos
  * @since 4.0.0
  */
-public interface AgentLauncher {
+public interface AgentLauncher extends HealthIndicator {
```

Comment:
I believe you need to update `genie.tasks.cluster-checker.healthIndicatorsToIgnore` with the new name that this indicator will get.
Otherwise the leader is gonna mark jobs on an host failed when that host is out of memory.

Reply:
I double checked by running the OSS one and comparing it to our current deployment. Both get the key `localAgentLauncher` so I think we're ok here.

Reply:
Weird... that's not the bean name, nor the class name, nor is a constant i could find anywhere. Kind of an incredible coincidence that they get the same name 🤷‍♂️

Reply:
Probably some convention where they rip off `Impl` or other `HealthIndicator` to get to the core class name. I tried looking for something in the API which would allow us to dictate the name but I couldn't find anything which was surprising.